js/reference.js

```diff
@@ -1839,7 +1843,26 @@
                 var config = {
                     headers: this._generateContextHeader(contextHeaderParams)
                 };
-                this._server._http.delete(this.location.ermrestCompactUri, config).then(function (deleteResponse) {
+
+                var uri = this.table.uri;
+
+                var keys = [];
+                //create key for each tuple and push it into the key array
+                for (var i=0; i<tuples.length; i++) {
+                    // NOTE: should probably be factored out into a function of table
+                    var keyPair = "";
+                    for (var j=0; j<this.table.shortestKey.length; j++) {
+                        var colName = this.table.shortestKey[j].name;
+                        keyPair += module._fixedEncodeURIComponent(colName) + "=" + module._fixedEncodeURIComponent(tuples[i][colName]);
```

Comment:
Since it's now an array of objects, I think we should check for tuples[i][colName] to be available. You might want to throw an error if it's not available.